File: NaturalEventsViewer/Eonet.Core/Models/EonetEvent.cs
﻿using System;

namespace Eonet.Core.Models
{
    public class EonetEvent
    {
        public string Id { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public string Link { get; set; }
        public DateTime? Closed { get; set; }
        public EonetCategory[] Categories { get; set; }
        public EonetSource[] Sources { get; set; }
        public EonetGeometry[] Geometry { get; set; }
    }
}

File: NaturalEventsViewer/NaturalEventsViewer.Client/ClientApp/shared/models/EonetEvent.ts
export type EonetEvent = {
    id: string;
    title: string;
    description: string | null;
    link: string;
    closedDate: string | null; // TODO: Use Date type instead
    lastDate: string; // TODO: Use Date type instead
    status: EonetEventStatus;
    category: EonetCategory;
    sources: EonetSource[];
    geometry: EonetGeometry[];
}

export enum EonetEventStatus {
    Open = 'Open',
    Closed = 'Closed'
}

export type EonetCategory = {
    id: string;
    title: string;
}

export type EonetSource = {
    id: string;
    url: string;
}

export type EonetGeometry = {
    date: string;
    type: EonetGeometryType;
}

export enum EonetGeometryType {
    Point = 'Point',
    Polygon = 'Polygon'
}

export type EonetGeometryPoint = EonetGeometry & {
    coordinates: number[];
}

export type EonetGeometryPolygon = {
    coordinates: number[][][];
}